cmd/thermal-recorder/snapshot.go

```diff
@@ -66,36 +65,45 @@ func newSnapshotRecording() error {
 	defer mu.Unlock()
 
 	if processor == nil {
+		log.Println("no motion processor so can't make snapshot")
 		return errors.New("reading from camera has not started yet")
 	}
 
 	processor.StartSnapshot = true
 	return nil
 }
 
-func snapshotRecordingTriggers(processor *motion.MotionProcessor, window window.Window) {
+// snapshotRecordingTriggers will make a snapshot when in the recording window and at the end of the recording window.
+func snapshotRecordingTriggers(window window.Window) {
+
+	// Wait for motion processor to start
+	for processor == nil {
+		time.Sleep(time.Second)
+	}
+
 	if window.NoWindow {
 		log.Println("no recording window so will make snapshot every 12 hours")
 		triggerTime := time.Now().Add(time.Minute)
 		for {
 			time.Sleep(time.Until(triggerTime))
-			log.Println("making snapshot")
-			processor.StartSnapshot = true
+			_ = newSnapshotRecording()
 			triggerTime = triggerTime.Add(time.Hour * 12)
 		}
 	}
 
-	// Make snapshot at start of window.
-	sleepTime := time.Minute
-	if !window.Active() {
-		sleepTime = time.Until(window.NextStart()) + time.Minute
+	if window.Active() {
+		// If camera just started give it a minute to warm  up.
+		time.Sleep(time.Minute)
+		log.Println("making power on snapshot")
```

Comment:
Is it possible we will miss snapshots sometimes as the camera model may take longer than a minute to connect?